src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -261,15 +262,25 @@ public void write() {
               }
             } else {
               SinkRecord projectedRecord = SchemaUtils.project(record, currentSchema, compatibility);
-              writeRecord(projectedRecord);
-              buffer.poll();
-              if (shouldRotate(now)) {
+              String partition = partitioner.encodePartition(projectedRecord);
+              if (currentPartition != null && !currentPartition.equals(partition)) {
+                currentPartition = partition;
```

Comment:
I see, that is not possible at the moment. I think you'd want something like `rotate.schedule.ms` that works a bit more like a cron specification than `rotate.interval.ms` which is just the interval but says nothing about starting time/alignment.